### ISSUE TITLE
aws_apigatewayv2_api - Add BodyS3Location

### ISSUE THREAD

**Issue by mcqj** [terraform-providers/terraform-provider-aws#13328]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The AWS V2 API for HTTP supports "BodyS3Location" - see Cloudformation docs https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-api.html
```
The S3 location of an OpenAPI definition. Supported only for HTTP APIs. To import an HTTP API, you must specify a Body or BodyS3Location.
```
### Potential Terraform Configuration

Would look like:-

```hcl
resource "aws_apigatewayv2_api" "example" {
  bodys3location = aws_s3_bucket.x.bucket
}
```

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/pull/12567

@mcqj Thanks for opening this issue.
The linked PR above implements the functionality required to import an API given a string body.
The underlying [AWS API used](https://docs.aws.amazon.com/sdk-for-go/api/service/apigatewayv2/#ReimportApiInput) only supports passing the body as a string.
CloudFormation must be getting the specified S3 object under the covers and passing it to the relevant API Gateway API.

Once the linked PR is merged you should be able ti use the [`aws_s3_bucket_object`](https://www.terraform.io/docs/providers/aws/d/s3_bucket_object.html) data source to obtain the S3 object's body and pass it as an input attribute to the `aws_apigatewayv2_api` resource.

**Comment by ewbankkit** [comment]
@mcqj #12567 has now been merged and will be released with v3.3.0 if the Terraform AWS Provider.
With the new `body` attribute on the `aws_apigatewayv2_api` resource you should be able to get the functionality you're looking for with some code like:

```hcl
data "aws_s3_bucket_object" "example" {
  bucket = "bucket-containing-api-spec"
  key    = "/path/to/api-spec.yaml"
}

resource "aws_apigatewayv2_api" "example" {
  name          = "example-http-api"
  protocol_type = "HTTP"

  body = data.aws_s3_bucket_object.example.body
}
```
